### PR TITLE
Use native ESM and update dependencies

### PR DESCRIPTION
**Pull request by niksy**
This PR will:

* Update package to be ESM only
* Update `gifsicle` to `^6.0.0` (this probably closes https://github.com/imagemin/imagemin-gifsicle/pull/47)
* Add meta tweaks

### CODE REVIEW

**Review by Diono — APPROVED** [review]

File: test.js [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
+import {Buffer} from 'node:buffer';
+import {promises as fs} from 'node:fs';
+import {fileURLToPath} from 'node:url';
+import isGif from 'is-gif';
+import test from 'ava';
+import imageminGifsicle from './index.js';
 
 test('Buffer', async t => {
-	const buf = await readFile(path.join(__dirname, 'fixture.gif'));
+	const buf = await fs.readFile(fileURLToPath(new URL('fixture.gif', import.meta.url)));
```
Comment (1000ch) [inline test.js:9]:
```suggestion
	const buf = await fs.readFile(new URL('fixture.gif', import.meta.url));
```

File: test.js [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-const fs = require('fs');
-const path = require('path');
-const isGif = require('is-gif');
-const test = require('ava');
-const imageminGifsicle = require('.');
-
-const readFile = promisify(fs.readFile);
+import {Buffer} from 'node:buffer';
+import {promises as fs} from 'node:fs';
+import {fileURLToPath} from 'node:url';
```
Comment (1000ch) [inline test.js:3]:
```suggestion
```

**Comment by niksy** [comment]
@1000ch everything applied!

File: readme.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# imagemin-gifsicle
+# imagemin-gifsicle ![GitHub Actions Status](https://github.com/imagemin/imagemin-gifsicle/workflows/test/badge.svg?branch=master)
```
Comment (1000ch) [inline readme.md:1]:
```suggestion
# imagemin-gifsicle ![GitHub Actions Status](https://github.com/imagemin/imagemin-gifsicle/workflows/test/badge.svg?branch=main)
```

**Pull request merged**